scripts/src/javatests/org/oppia/android/scripts/license/MavenDependenciesRetrieverTest.kt

```diff
@@ -0,0 +1,1312 @@
+      )
+    )
+    assertThat(mavenListDependencies).doesNotContain(
+      MavenListDependency(
+        coord = DEP_WITH_INVALID_LINKS,
+        url = "${IO_FABRIC_POM.dropLast(3)}aar"
+      )
+    )
+  }
+
+  private fun verifyLicenseHasScrapableVerifiedLink(
+    license: License,
+    originalLink: String,
+    licenseName: String,
+    verifiedLink: String,
+  ) {
+    assertThat(license.licenseName).isEqualTo(licenseName)
+    assertThat(license.verifiedLinkCase).isEqualTo(
+      License.VerifiedLinkCase.SCRAPABLE_LINK
+    )
+    assertThat(license.scrapableLink.url).isEqualTo(verifiedLink)
+    assertThat(license.originalLink).isEqualTo(originalLink)
+    assertThat(license.isOriginalLinkInvalid).isFalse()
+  }
+
+  private fun verifyLicenseHasExtractedCopyVerifiedLink(
+    license: License,
+    originalLink: String,
+    licenseName: String,
+    verifiedLink: String,
+  ) {
+    assertThat(license.licenseName).isEqualTo(licenseName)
+    assertThat(license.verifiedLinkCase).isEqualTo(
+      License.VerifiedLinkCase.EXTRACTED_COPY_LINK
+    )
+    assertThat(license.extractedCopyLink.url).isEqualTo(verifiedLink)
+    assertThat(license.originalLink).isEqualTo(originalLink)
+    assertThat(license.isOriginalLinkInvalid).isFalse()
+  }
+
+  private fun verifyLicenseHasDirectLinkOnlyVerifiedLink(
+    license: License,
+    originalLink: String,
+    licenseName: String,
+    verifiedLink: String,
+  ) {
+    assertThat(license.licenseName).isEqualTo(licenseName)
+    assertThat(license.verifiedLinkCase).isEqualTo(
+      License.VerifiedLinkCase.DIRECT_LINK_ONLY
+    )
+    assertThat(license.directLinkOnly.url).isEqualTo(verifiedLink)
+    assertThat(license.originalLink).isEqualTo(originalLink)
+    assertThat(license.isOriginalLinkInvalid).isFalse()
+  }
+
+  private fun verifyLicenseHasVerifiedLinkNotSet(
+    license: License,
+    originalLink: String,
+    licenseName: String
+  ) {
+    assertThat(license.licenseName).isEqualTo(licenseName)
+    assertThat(license.verifiedLinkCase).isEqualTo(License.VerifiedLinkCase.VERIFIEDLINK_NOT_SET)
+    assertThat(license.originalLink).isEqualTo(originalLink)
+    assertThat(license.isOriginalLinkInvalid).isFalse()
+  }
+
+  private fun verifyLicenseHasOriginalLinkInvalid(
+    license: License,
+    originalLink: String,
+    licenseName: String
+  ) {
+    assertThat(license.licenseName).isEqualTo(licenseName)
+    assertThat(license.verifiedLinkCase).isEqualTo(License.VerifiedLinkCase.VERIFIEDLINK_NOT_SET)
+    assertThat(license.originalLink).isEqualTo(originalLink)
+    assertThat(license.isOriginalLinkInvalid).isTrue()
+  }
+
+  private fun assertIsDependency(
+    dependency: MavenDependency,
+    artifactName: String,
+    artifactVersion: String,
+  ) {
+    assertThat(dependency.artifactName).isEqualTo(artifactName)
+    assertThat(dependency.artifactVersion).isEqualTo(artifactVersion)
+  }
+
+  private fun parseTextProto(
```

Comment:
This appears to be unused, but it probably should be per my earlier comment.

That being said, please make sure to remove all unused variables/members.

Reply:
Done.